include/veriblock/blockchain/pop/pop_state_machine.hpp

```diff
@@ -202,6 +203,7 @@ struct PopStateMachine {
     }
 
     index.unsetFlag(BLOCK_APPLIED);
+    --ed_.appliedBlockCount;
```

Comment:
I'd add an assert that appliedBlockCount is not zero.

Reply:
there's a more elaborate check in setState() that would also catch this case, but it doesn't hurt, yeah